ts/Grid/Core/Credits.ts

```diff
@@ -200,7 +163,25 @@ class Credits {
  * */
 
 namespace Credits {
+    /**
+     * Extends the grid classes with credits.
+     *
+     * @param GridClass
+     * The class to extend.
+     *
+     */
+    export function compose(
+        GridClass: typeof Grid
+    ): void {
+        addEvent(GridClass, 'afterRenderViewport', initCredits);
+    }
 
+    /**
+     * Callback function called before table initialization.
+     */
+    function initCredits(this: Grid): void {
+        new Credits(this, Credits.defaultOptions);
```

Comment:
Instead of passing here `Credits.defaultOptions`, I would pass here the options directly. They could be easily overwritten in `Credits.defaultOptions`. Or I would define it in the non-exported global const, outside the namespace or class block.